cmd/gitserver/server/customfetch.go

```diff
@@ -2,19 +2,55 @@ package server
 
 import (
 	"context"
+	"net/url"
 	"os/exec"
+	"path"
 	"strings"
 
-	"github.com/sourcegraph/sourcegraph/internal/env"
+	"github.com/sourcegraph/sourcegraph/internal/conf"
+	"github.com/sourcegraph/sourcegraph/schema"
+	"gopkg.in/inconshreveable/log15.v2"
 )
 
-var customFetch = strings.Fields(env.Get("SRC_GITSERVER_CUSTOM_FETCH", "",
-	"EXPERIMENTAL: custom fetch command for unorthodox repo setups."))
+var customGitFetch = conf.Cached(func() interface{} {
+	return buildCustomFetchMappings(conf.Get().ExperimentalFeatures.CustomGitFetch)
+})
 
-func useCustomFetch() bool {
-	return len(customFetch) > 0
+func buildCustomFetchMappings(c []*schema.CustomGitFetchMapping) map[string][]string {
+	if c == nil {
+		return map[string][]string{}
+	}
+
+	cgm := map[string][]string{}
+	for _, mapping := range c {
+		parts := strings.Fields(mapping.Fetch)
+		cgm[mapping.DomainPath] = parts
+	}
+
+	return cgm
 }
 
-func customFetchCmd(ctx context.Context) *exec.Cmd {
-	return exec.CommandContext(ctx, customFetch[0], customFetch[1:]...)
+func extractDomainPath(cloneURL string) (string, error) {
+	gitURL, err := url.Parse(cloneURL)
+	if err != nil {
+		return "", err
+	}
+
+	return path.Join(gitURL.Host, gitURL.Path), nil
+}
+
+func customFetchCmd(ctx context.Context, urlVal string) *exec.Cmd {
+	cgm := customGitFetch().(map[string][]string)
+
+	dp, err := extractDomainPath(urlVal)
+	if err != nil {
+		log15.Error("failed to extract domain and path from %s", urlVal)
```

Comment:
You forgot the most important part :)
```suggestion
		log15.Error("failed to extract domain and path from %s: %v", urlVal, err)
```